eldritch/events.py

```diff
@@ -3144,6 +3164,33 @@ def annotations(self, state):
     return None
 
 
+class OverrideGateChoice(Event):
+  def __init__(self, character, original_choice: GateChoice, **changes):
+    super().__init__()
+    self.character = character
+    self.original_choice = original_choice
+    self.changes = changes
+    assert len(changes)
+    assert {"prompt", "gate_name", "none_choice", "annotation"}.issuperset(changes.keys())
+    self.done = False
+
+  def resolve(self, state):
+    for key, value in self.changes.items():
+      setattr(self.original_choice, key, value)
+    self.original_choice.compute_choices(state)
```

Comment:
Removing it causes `testTradeBeforeReturnMultipleGates` to fail at 
```
      self.assertEqual(nun_choice.character, nun)
>     self.assertSequenceEqual(nun_choice.choices, ["Diner", "Woods"])
E     AssertionError: Sequences differ: ['Woods'] != ['Diner', 'Woods']
E
E     First differing element 0:
E     'Woods'
E     'Diner'
E
E     Second sequence contains 1 additional elements.
E     First extra element 1:
E     'Woods'
```

Reply:
So I tried this out - it looks like there are two things going on here. The first is that after Dummy hands the GateBox to the Nun, he is still able to trade - you must set done_using[0] to True to continue the test normally.

The test currently expects the GateBox to be usable for the Nun. But this is only happening because the Dummy hasn't successfully returned. Once you fix the issue above and the Dummy successfully returns, the GateBox will be unconditional and the test will pass.

But this test has uncovered a second issue. You see, compute_choices for the GateChoice has a special side effect when there is only one choice - it will automatically choose the return destination. This was implemented based on feedback I received when we were last testing it out. So in this case, the GateChoice will compute choices, see that there is only one choice, and set event.choice = the only choice. Then, the GateBox comes along and changes event.gate_name to None. But the event loop sees the GateChoice as already being done, so it doesn't bother to call compute_choices again.

Stick a TODO in there for me to fix. It's a mess that I need to untangle.